discovery/service.go

```diff
@@ -526,12 +627,219 @@ func (d *Discovery) processNetworkAnnouncement(aMsg *networkMsg) bool {
 		}
 
 		if err := d.cfg.Router.UpdateEdge(update); err != nil {
-			log.Errorf("unable to update edge: %v", err)
-			return false
+			e := errors.Errorf("unable to update edge: %v", err)
+			if routing.IsError(err,
+				routing.ErrOutdated,
+				routing.ErrIgnored) {
+				log.Info(e)
+			} else {
+				log.Error(e)
+			}
+
+			nMsg.err <- e
+			return nil
 		}
-	}
 
-	return true
+		if !nMsg.isRemote {
+			remotePubKey := oppositePubKey
+			d.cfg.SendToPeer(remotePubKey, msg)
```

Comment:
Because `ChannelUpdateAnnouncement` message shouldn't be broadcasted if we are not going to announce the channel, but it should be sent to the remote side in order to consider this channel to be opened.